python/ray/autoscaler/updater.py

```diff
@@ -316,6 +316,32 @@ def run(self,
             port_forward=None,
             with_output=False)
 
+    def ensure_docker_installed(self):
+        try:
+            self.ssh_command_runner.run("command -v docker")
+            return
+        except Exception:
+            logger.info("Docker not installed, installing now")
+
+        # Disable session reuse to allow for group changes to be
+        # correctly reflected.
+        old_opts = self.ssh_command_runner.OPTS
+        new_opts = [opt for opt in old_opts if "Control" not in opt[0]]
+        self.ssh_command_runner.OPTS = new_opts
```

Comment:
this code is actually quite hard to reason about - I don't immediately know what the state of OPTS is. 

Can we please avoid doing this and just manually override it via `run`?

Reply:
add an argument to `run` that allows special OPTS. or manually call subprocess with `ssh`?

Reply:
yes! use adding args sgtm